server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -128,16 +144,27 @@ JavaDateFormatter getRoundupParser() {
         }
         this.printer = printer;
         this.format = format;
+        this.printFormat = printFormat;
+        this.canCacheLastParsedFormatter = canCacheLastParsedFormatter;
 
         if (parsers.length == 0) {
             this.parsers = Collections.singletonList(printer);
         } else {
-            this.parsers = Arrays.asList(parsers);
+            this.parsers = new CopyOnWriteArrayList<>(parsers);
```

Comment:
@CaptainDredge since access is synchronized now, we don't need `CopyOnWriteArrayList` anymore

Reply:
We'd still need CopyOnWriteArrayList for iterating on the list, it'll provide a snapshot of the list to iterate. Multiple threads can iterate and parse this way.

Reply:
The iteration won't fail with `ConcurrentModificationException` but it may say iterate over partially updated list of the parsers (https://github.com/opensearch-project/OpenSearch/pull/9567#discussion_r1346640968)